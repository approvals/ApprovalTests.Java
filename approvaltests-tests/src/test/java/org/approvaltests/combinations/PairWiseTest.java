package org.approvaltests.combinations;

import org.junit.jupiter.api.Test;

public class PairWiseTest {

    @Test
    public void testPairs()
    {
        PairWiseApprovals.verifyBestCoveringPairs((a,b,c,d) -> "",
                new Integer[]{1, 2, 3, 4, 5},
                new String[]{"a", "b", "c", "d"},
                new String[]{"L","M","N","O","P"},
                new Double[]{1.1,2.2,3.3,4.4});
    }

}
