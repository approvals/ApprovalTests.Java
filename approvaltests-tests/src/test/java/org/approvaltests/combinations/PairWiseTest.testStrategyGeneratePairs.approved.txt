Generate Pairs[0] = []
Generate Pairs[1] = [{p2=1, p5=1.1}, {p2=2, p5=1.1}, {p2=3, p5=1.1}, {p2=1, p5=2.2}, {p2=2, p5=2.2}, {p2=3, p5=2.2}, {p2=1, p5=3.3}, {p2=2, p5=3.3}, {p2=3, p5=3.3}]
Generate Pairs[2] = [{p2=1, p3=a}, {p2=2, p3=a}, {p2=3, p3=a}, {p5=1.1, p3=a}, {p5=2.2, p3=a}, {p5=3.3, p3=a}, {p2=1, p3=b}, {p2=2, p3=b}, {p2=3, p3=b}, {p5=1.1, p3=b}, {p5=2.2, p3=b}, {p5=3.3, p3=b}]
Generate Pairs[3] = [{p2=1, p4=L}, {p2=2, p4=L}, {p2=3, p4=L}, {p5=1.1, p4=L}, {p5=2.2, p4=L}, {p5=3.3, p4=L}, {p3=a, p4=L}, {p3=b, p4=L}, {p2=1, p4=M}, {p2=2, p4=M}, {p2=3, p4=M}, {p5=1.1, p4=M}, {p5=2.2, p4=M}, {p5=3.3, p4=M}, {p3=a, p4=M}, {p3=b, p4=M}]
